site/blog/2022/deploy-ghost-ec2.mdx

```diff
@@ -0,0 +1,193 @@
+---
+slug: deploy-ghost-ec2
+title: Deploy Ghost in EC2 with one click
+date: 2022-11-23
+authors: [yrobla]
+---
+
+# Deploy <a href="https://ghost.org/">Ghost</a> with one click
+
+Ghost is a powerful app for new-media creators to publish, share, and grow a business around their content. It comes with modern tools to build a website, publish content, send newsletters & offer paid
+subscriptions to members.
+
+Ghost is a software as a service product, but it is totally built on open source and offers a self-hosted, installable version. This version can be hosted on AWS using several approaches, and one of them
+is to host on an EC2 instance.
+
+Starting an EC2 instance and manually triggering the installation commands is a procedure that can be done manually, but there are also another dependencies involved, such as the creation of proper security groups.
+It is time-consuming task, specially if you need to install several instances. This article will offer a way to install Ghost with one-click on IaSQL using an EC2 instance on Ubuntu as highlighted <a href="https://ghost.org/docs/install/ubuntu/">here</a>.
+
+```sql TheButton[Install Ghost]="Install Ghost"
+ -- Install modules
+SELECT
+  *
+FROM
+  iasql_install ('aws_ec2', 'aws_ec2_metadata');
+
+-- insert security groups
+INSERT INTO
+  security_group (description, group_name)
+VALUES
+  ('Ghost security group', 'ghost_sg_group');
+
+INSERT INTO
+  security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+SELECT
+  FALSE,
+  'tcp',
+  22,
+  22,
+  '0.0.0.0/0',
+  'ghost_sg_group_rule_ssh',
+  id
+FROM
+  security_group
+WHERE
+  group_name = 'ghost_sg_group';
+
+INSERT INTO
+  security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+SELECT
+  FALSE,
+  'tcp',
+  80,
+  80,
+  '0.0.0.0/0',
+  'ghost_sg_group_rule_http',
+  id
+FROM
+  security_group
+WHERE
+  group_name = 'ghost_sg_group';
+
+INSERT INTO
+  security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+SELECT
+  FALSE,
+  'tcp',
+  443,
+  443,
+  '0.0.0.0/0',
+  'ghost_sg_group_rule_https',
+  id
+FROM
+  security_group
+WHERE
+  group_name = 'ghost_sg_group';
+
+INSERT INTO
+  security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+SELECT
+  TRUE,
+  'tcp',
+  1,
+  65335,
+  '0.0.0.0/0',
+  'ghost_sg_group_rule_egress',
+  id
+FROM
+  security_group
+WHERE
+  group_name = 'ghost_sg_group';
```

Comment:
This is super repetitive. Postgres' [`values` syntax](https://www.postgresql.org/docs/current/sql-values.html) is more advanced than some other sql engines, so I think we can do something like:

```sql
INSERT INTO
  security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
SELECT
  t.is_egress,
  t.ip_protocol,
  t.from_port,
  t.to_port,
  t.cidr_ipv4,
  t.description,
  security_group.id
FROM
  security_group, (VALUES
    (FALSE, 'tcp', 22, 22, '0.0.0.0/0', 'ghost_sg_group_rule_ssh'),
    (FALSE, 'tcp', 80, 80, '0.0.0.0/0', 'ghost_sg_group_rule_http'),
    (FALSE, 'tcp', 443, 443, '0.0.0.0/0', 'ghost_sg_group_rule_https'),
    (TRUE, 'tcp', 1, 65535, '0.0.0.0/0', 'ghost_sg_group_rule_egress')
  ) as t (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description)
WHERE
  security_group.group_name = 'ghost_sg_group';
```